### PR TITLE
[RFC] build/msvc: Fix functional tests

### PR DESCRIPTION
**Pull request by b-r-o-c-k**
All the functional tests are passing, but MSBuild still returns a non-zero exit code because it detects the word "error" in the stdout which is caused by some of the test names such as `api/buf {get,set,del}_line get_line : out-of-bounds is an error`. There is a [CMake mailing list thread](https://cmake.org/pipermail/cmake-developers/2015-October/026775.html) about this problem, but there isn't any good solution for it, so I modified the build script to detect the error message printed by `RunTests.cmake`.

### CODE REVIEW

File: ci/build.ps1 [hunk 1]
```diff
@@ -11,7 +11,6 @@ $depsCmakeVars = @{
 $nvimCmakeVars = @{
   CMAKE_BUILD_TYPE = $cmakeBuildType;
   BUSTED_OUTPUT_TYPE = 'nvim';
-  GPERF_PRG = 'C:\msys64\usr\bin\gperf.exe';
```
Comment (b-r-o-c-k) [inline ci/build.ps1]: This isn't needed anymore now that gperf is part of the bundled dependencies.

**Comment by janlazo** [comment]
Does this CMake issue apply to oldtests?

**Comment by b-r-o-c-k** [comment]
@janlazo No, the oldtests don't use CMake so it shouldn't be a problem.

**Comment by justinmk** [comment]
Thanks @b-r-o-c-k . I added your explanation to the merge-commit message, but generally that info should be included in the specific related commit (unless it's really not useful outside the contemporaneous context of a PR discussion).

**Pull request merged**